app/actions/local/session/index.ts

```diff
@@ -104,24 +104,87 @@ export const cancelSessionNotification = async (serverUrl: string) => {
 };
 
 export const terminateSession = async (serverUrl: string, removeServer: boolean) => {
-    cancelSessionNotification(serverUrl);
-    await removeServerCredentials(serverUrl);
+    const errors: Array<{operation: string; error: unknown}> = [];
 
+    // Helper to safely execute operations and optionally track errors
+    const safeExecute = async (operation: string, fn: () => Promise<unknown>, critical = true) => {
```

Comment:
I like this helper, should it be outside of the terminatesession function so we can add tests to it?

Extraction and tests are not needed in this PR, but it would be great to have a ticket for it :)